jetty-server/src/main/java/org/eclipse/jetty/server/CookieCutter.java

```diff
@@ -269,34 +316,19 @@ else if(tokenstart>=0 && tokenend>=0)
                             case '\t':
                                 continue;
 
-                            case ';':
-                                if (quoted)
-                                {
-                                    name = unquoted.toString();
-                                    unquoted.setLength(0);
-                                    quoted = false;
-                                }
-                                else if(tokenstart>=0 && tokenend>=0)
-                                {
-                                    name = hdr.substring(tokenstart, tokenend+1);
-                                }
-
-                                tokenstart = -1;
-                                break;
-
                             case '=':
                                 if (quoted)
                                 {
                                     name = unquoted.toString();
                                     unquoted.setLength(0);
                                     quoted = false;
                                 }
-                                else if(tokenstart>=0 && tokenend>=0)
-                                {
-                                    name = hdr.substring(tokenstart, tokenend+1);
-                                }
+                                else if(tokenstart>=0)
+                                    name = tokenend>=tokenstart?hdr.substring(tokenstart, tokenend+1):hdr.substring(tokenstart);
+
                                 tokenstart = -1;
-                                invalue=true;
+                                if (c=='=')
```

Comment:
Why is this if needed? It seems redundant to me.

Reply:
Ah true... at one stage I was handling '0' and ';' name termination... but no longer needed